src/main/kotlin/solve/importer/ProjectParser.kt

```diff
@@ -154,55 +160,76 @@ object ProjectParser {
 
         for (folder in directory.listFiles()) {
             if (folder.name != IMAGE_DIRECTORY_NAME) {
-                parseOutputs(folder, layers, landmarks, errorFolders)
+                partialParseOutputs(folder, outputs, errorFolders, algorithms, layers)
             } else {
                 isImagesExist = true
-                parseImages(folder, images)
+                partialParseImages(folder, images)
             }
         }
 
         if (!isImagesExist) {
+            val ownerWindow = importer.root.scene.window
             createAlertForError("The images folder is missing in the directory", ownerWindow)
             return null
         }
 
+        images.map { img ->
+            if (img.errors.isNotEmpty()) {
+                hasAnyErrors = true
+            }
+            frames.add(FrameAfterPartialParsing(img, outputs.filter { img.name == it.name }))
+        }
+
+        if (noSomeAlgorithmError(frames, layers)) {
+            hasAnyErrors = true
+        }
+
+        frames.sortWith(compareBy { it.image.name.toLong() })
```

Comment:
Why it is important?

Reply:
so that the images are displayed in the correct order